### PR TITLE
[aws-greengrass-core-sdk-c] Add new port

### PR DESCRIPTION
**Pull request by pas-mike**
SDK to use with functions running on Greengrass Core using C

Not supported for windows.

### CODE REVIEW

File: ports/aws-greengrass-core-sdk-c/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#vcpkg_fixup_cmake_targets(CONFIG_PATH lib/aws-greengrass-core-sdk-c/cmake)
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake)
+
+file(REMOVE_RECURSE
+    ${CURRENT_PACKAGES_DIR}/debug/include
+    ${CURRENT_PACKAGES_DIR}/debug/share
+	${CURRENT_PACKAGES_DIR}/debug/lib/${PORT}
+	${CURRENT_PACKAGES_DIR}/lib/${PORT}
+)
+
```
Comment (NancyLi1013) [inline ports/aws-greengrass-core-sdk-c/portfile.cmake]: Could you please update the format as this?
```
file(REMOVE_RECURSE
    ${CURRENT_PACKAGES_DIR}/debug/include
    ${CURRENT_PACKAGES_DIR}/debug/share
    ${CURRENT_PACKAGES_DIR}/debug/lib/${PORT}
    ${CURRENT_PACKAGES_DIR}/lib/${PORT}
)
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @PAMike thanks for this PR.
I noticed this port failed on osx pipeline. 
Could you please take a look and try to fix the regression?
Also, since this port doesn't support Windows platform, please update [baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) with the following codes:
```
aws-greengrass-core-sdk-c:x86-windows=fail
aws-greengrass-core-sdk-c:x64-windows=fail
aws-greengrass-core-sdk-c:x64-windows-static=fail
aws-greengrass-core-sdk-c:x64-uwp=fail
aws-greengrass-core-sdk-c:arm64-windows=fail
aws-greengrass-core-sdk-c:arm-uwp=fail
```
What's more, I noticed that this PR contains `mosquitto` update.
I saw you have submitted a PR #9639 to update `mosquitto`.
So please revert the commits about `mosquitto`.

**Comment by NancyLi1013** [comment]
Could you resolve the conflicts?

**Comment by NancyLi1013** [comment]
The regression on osx pipeline:
```
[8/17] : && /Library/Developer/CommandLineTools/usr/bin/cc -fPIC -g -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -dynamiclib -Wl,-headerpad_max_install_names -Wl,--version-script=/Users/vagrant/azure-agent/_work/2/s/buildtrees/aws-greengrass-core-sdk-c/src/v1.2.0-319fa23e6e/aws-greengrass-core-sdk-c/version/greengrasssdk_versions.map -o aws-greengrass-core-sdk-c/libaws-greengrass-core-sdk-c.dylib -install_name @rpath/libaws-greengrass-core-sdk-c.dylib aws-greengrass-core-sdk-c/CMakeFiles/aws-greengrass-core-sdk-c.dir/lib/greengrasssdk.c.o  -Wl,--enable-new-dtags && :
FAILED: aws-greengrass-core-sdk-c/libaws-greengrass-core-sdk-c.dylib 
: && /Library/Developer/CommandLineTools/usr/bin/cc -fPIC -g -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -dynamiclib -Wl,-headerpad_max_install_names -Wl,--version-script=/Users/vagrant/azure-agent/_work/2/s/buildtrees/aws-greengrass-core-sdk-c/src/v1.2.0-319fa23e6e/aws-greengrass-core-sdk-c/version/greengrasssdk_versions.map -o aws-greengrass-core-sdk-c/libaws-greengrass-core-sdk-c.dylib -install_name @rpath/libaws-greengrass-core-sdk-c.dylib aws-greengrass-core-sdk-c/CMakeFiles/aws-greengrass-core-sdk-c.dir/lib/greengrasssdk.c.o  -Wl,--enable-new-dtags && :
ld: unknown option: --version-script=/Users/vagrant/azure-agent/_work/2/s/buildtrees/aws-greengrass-core-sdk-c/src/v1.2.0-319fa23e6e/aws-greengrass-core-sdk-c/version/greengrasssdk_versions.map
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
Could you take a look and try to fix it?

**Comment by NancyLi1013** [comment]
Could you please resolve the conflicts?

**Comment by NancyLi1013** [comment]
Pinging @PAMike Is work still being done for this PR?

**Comment by fsmikedev** [comment]
Hi @NancyLi1013, sorry for not responding a lot sooner, I need to investigate the OSX build.

**Comment by NancyLi1013** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being down.

**Pull request closed without merging**